pkg/cmd/pr/checks/checks.go

```diff
@@ -0,0 +1,211 @@
+	if err != nil {
+		return err
+	}
+
+	if len(pr.Commits.Nodes) == 0 {
+		return nil
+	}
+
+	rollup := pr.Commits.Nodes[0].Commit.StatusCheckRollup.Contexts.Nodes
+	if len(rollup) == 0 {
+		return nil
+	}
+
+	passing := 0
+	failing := 0
+	pending := 0
+
+	type output struct {
+		mark    string
+		bucket  string
+		name    string
+		elapsed string
+		link    string
+	}
+
+	outputs := []output{}
+
+	for _, c := range pr.Commits.Nodes[0].Commit.StatusCheckRollup.Contexts.Nodes {
+		mark := ""
+		bucket := ""
+		state := c.State
+		if state == "" {
+			if c.Status == "COMPLETED" {
+				state = c.Conclusion
+			} else {
+				state = c.Status
+			}
+		}
+		switch state {
+		case "SUCCESS", "NEUTRAL", "SKIPPED":
+			mark = utils.GreenCheck()
+			passing++
+			bucket = "pass"
+		case "ERROR", "FAILURE", "CANCELLED", "TIMED_OUT", "ACTION_REQUIRED":
+			mark = utils.RedX()
+			failing++
+			bucket = "fail"
+		case "EXPECTED", "REQUESTED", "QUEUED", "PENDING", "IN_PROGRESS", "STALE":
+			mark = utils.YellowDash()
+			pending++
+			bucket = "pending"
+		default:
+			panic(fmt.Errorf("unsupported status: %q", state))
+		}
+
+		elapsed := ""
+		zeroTime := time.Time{}
+
+		if c.StartedAt != zeroTime && c.CompletedAt != zeroTime {
+			e := c.CompletedAt.Sub(c.StartedAt)
+			if e > 0 {
+				elapsed = e.String()
+			}
+		}
+
+		link := c.DetailsURL
+		if link == "" {
+			link = c.TargetURL
+		}
+
+		name := c.Name
+		if name == "" {
+			name = c.Context
+		}
+
+		outputs = append(outputs, output{mark, bucket, name, elapsed, link})
+	}
+
+	sort.Slice(outputs, func(i, j int) bool {
+		if outputs[i].bucket == outputs[j].bucket {
+			return outputs[i].name < outputs[j].name
+		} else {
+			if outputs[i].bucket == "fail" {
+				return true
+			} else if outputs[i].bucket == "pending" && outputs[j].bucket == "success" {
+				return true
+			}
+		}
+
+		return false
+	})
+
+	tp := utils.NewTablePrinter(opts.IO)
+
+	for _, o := range outputs {
+		if opts.IO.IsStdoutTTY() {
+			tp.AddField(o.mark, nil, nil)
+			tp.AddField(o.name, nil, nil)
+			tp.AddField(o.elapsed, nil, nil)
+			tp.AddField(o.link, nil, nil)
+		} else {
+			tp.AddField(o.name, nil, nil)
+			tp.AddField(o.bucket, nil, nil)
+			if o.elapsed == "" {
+				tp.AddField("0", nil, nil)
+			} else {
+				tp.AddField(o.elapsed, nil, nil)
+			}
+			tp.AddField(o.link, nil, nil)
+		}
+
+		tp.EndRow()
+	}
+
+	summary := ""
+	if failing+passing+pending > 0 {
+		if failing > 0 {
+			summary = "Some checks were not successful"
+		} else if pending > 0 {
+			summary = "Some checks are still pending"
+		} else {
+			summary = "All checks were successful"
+		}
+
+		tallies := fmt.Sprintf(
+			"%d failing, %d successful, and %d pending checks",
+			failing, passing, pending)
+
+		summary = fmt.Sprintf("%s\n%s", utils.Bold(summary), tallies)
+	}
+
+	if opts.IO.IsStdoutTTY() {
+		fmt.Fprintln(opts.IO.Out, summary)
+		fmt.Fprintln(opts.IO.Out)
+	}
+
+	return tp.Render()
```

Comment:
I forgot to bring this up before, but I feel that it's important that the `checks` command exits with a non-zero status if any checks are failing or still pending. That way, a script can run `gh pr checks >/dev/null` as a prerequisite for e.g. merging or deployment and have it fail the script when not all checks were successful.

Reply:
great idea.